src/transactions/simulation/TxSimUtils.cpp

```diff
@@ -212,10 +213,51 @@ mutateScaledAccountID(MuxedAccount& acc, uint32_t partition)
     return key;
 }
 
+PoolID
+getPoolID(LiquidityPoolConstantProductParameters const& params)
+{
+    LiquidityPoolParameters lpp(LIQUIDITY_POOL_CONSTANT_PRODUCT);
+    lpp.constantProduct() = params;
+    return xdrSha256(lpp);
+}
+
+PoolID
+getScaledPoolID(AbstractLedgerTxn& ltx, PoolID const& poolID,
+                UnorderedMap<PoolID, LiquidityPoolParameters>& ctPoolIdToParam,
+                uint32_t partition)
+{
+    auto pool = stellar::loadLiquidityPool(ltx, poolID);
+    if (!pool)
+    {
+        // This pool may be created in the same ledger
+        auto it = ctPoolIdToParam.find(poolID);
+        if (it != ctPoolIdToParam.end())
+        {
+            auto params = it->second;
+            replaceIssuer(params.constantProduct().assetA, partition);
+            replaceIssuer(params.constantProduct().assetB, partition);
+            return xdrSha256(params);
```

Comment:
same thing here, let's use the same `getPoolID` functionality everywhere. The same sequence of events happens later in this function on line 247, so it also can be extracted into a function

Reply:
Done